src/dev/create-extension.ts

```diff
@@ -0,0 +1,141 @@
+import arg from "arg";
+import path from "path";
+import fs from "fs";
+import { promisify } from "util";
+import { execa } from "execa";
+import ncp from "ncp";
+import { fileURLToPath } from "url";
+import { SOLIDITY_FRAMEWORKS } from "../utils/consts";
+import chalk from "chalk";
+
+const EXTERNAL_EXTENSIONS_DIR = "externalExtensions";
+const TARGET_EXTENSION_DIR = "extension";
+const TEMPLATE_FILE_SUFFIX = ".template.mjs";
+const BASE_PATH = "base";
+const SOLIDITY_FRAMEWORKS_PATH = "solidity-frameworks";
+const DEPLOYED_CONTRACTS_FILE = "deployedContracts.ts";
+const YARN_LOCK_FILE = "yarn.lock";
+
+const prettyLog = {
+  info: (message: string, indent = 0) => console.log(chalk.cyan(`${"  ".repeat(indent)}${message}`)),
+  success: (message: string, indent = 0) => console.log(chalk.green(`${"  ".repeat(indent)}✔ ${message}`)),
+  warning: (message: string, indent = 0) => console.log(chalk.yellow(`${"  ".repeat(indent)}⚠ ${message}`)),
+  error: (message: string, indent = 0) => console.log(chalk.red(`${"  ".repeat(indent)}✖ ${message}`)),
+};
+
+const ncpPromise = promisify(ncp);
+
+const currentFileUrl = import.meta.url;
+const templateDirectory = path.resolve(decodeURI(fileURLToPath(currentFileUrl)), "../../templates");
+
+const parseArguments = (rawArgs: string[]) => {
```

Comment:
looks like better to rename to `getProjectPath`

Reply:
Updated at https://github.com/scaffold-eth/create-eth/pull/73/commits/ad55ab142aad7d0979ef9af16fe43ad85b0a9e07